File: ts/test/dcp/test-dcp-2.ts
import test, {Macro} from 'ava';
import {todo} from '../../src/dcp/dcp-2';

const cases: boolean[] = [false];

const macro: Macro<[boolean]> = (t, expected) => {
	t.is(todo(), expected);
};

macro.title = (i: string | undefined): string => `todo()_${i ?? ''}`;

for (const [i, expected] of cases.entries()) {
	test(i.toString(), macro, expected);
}

File: rb/test/dcp/test_dcp_2.rb
# typed: false
# frozen_string_literal: true

require "config"
require "dcp/dcp_2"

module DCP
  class TestDCP2 < Minitest::Test
    include DCP2

    [
      false
    ].each.with_index do |expected, i|
      define_method(:"test_todo_#{i}") do
        assert_equal(expected, todo)
      end
    end
  end
end
